index.rst

```diff
@@ -1,51 +1,1166 @@
+    #include "ndarray/pybind11.h"
+    #include "ndarray/converter.h"
+
+    #include "lsst/afw/math/interpolate.h"
+
+    namespace py = pybind11;
+    using namespace pybind11::literals;
+
+    using namespace lsst::afw::math;
+
+    PYBIND11_DECLARE_HOLDER_TYPE(MyType, std::shared_ptr<MyType>);
+
+    PYBIND11_PLUGIN(_interpolate) {
+        py::module mod("_interpolate", "Python wrapper for afw _interpolate library");
+
+        if (_import_array() < 0) {
+                PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+                return nullptr;
+            }
+
+        mod.def("makeInterpolate", 
+                           (PTR(Interpolate) (*)(std::vector<double> const &,
+                                                 std::vector<double> const &,
+                                                 Interpolate::Style const)) makeInterpolate,
+                           "x"_a, "y"_a, "style"_a=Interpolate::AKIMA_SPLINE);
+        mod.def("makeInterpolate", 
+                           (PTR(Interpolate) (*)(ndarray::Array<double const, 1> const &,
+                                                 ndarray::Array<double const, 1> const &y,
+                                                 Interpolate::Style const)) makeInterpolate,
+                           "x"_a, "y"_a, "style"_a=Interpolate::AKIMA_SPLINE);
+        /* Module level */
+
+        /* Member types and enums */
+
+        /* Constructors */
+
+        /* Operators */
+
+        /* Members */
+        
+        py::class_<Interpolate, std::shared_ptr<Interpolate>> clsInterpolate(mod, "Interpolate");
+        py::enum_<Interpolate::Style>(clsInterpolate, "Style")
+            .value("UNKNOWN", Interpolate::Style::UNKNOWN)
+            .value("CONSTANT", Interpolate::Style::CONSTANT)
+            .value("LINEAR", Interpolate::Style::LINEAR)
+            .value("NATURAL_SPLINE", Interpolate::Style::NATURAL_SPLINE)
+            .value("CUBIC_SPLINE", Interpolate::Style::CUBIC_SPLINE)
+            .value("CUBIC_SPLINE_PERIODIC", Interpolate::Style::CUBIC_SPLINE_PERIODIC)
+            .value("AKIMA_SPLINE", Interpolate::Style::AKIMA_SPLINE)
+            .value("AKIMA_SPLINE_PERIODIC", Interpolate::Style::AKIMA_SPLINE_PERIODIC)
+            .value("NUM_STYLES", Interpolate::Style::NUM_STYLES)
+            .export_values();
+
+        clsInterpolate.def("interpolate", [](Interpolate &t, double const x) -> double {
+                return t.interpolate(x);
+        });
+        clsInterpolate.def("interpolate",
+                           (std::vector<double> (Interpolate::*) (std::vector<double> const&) const)
+                               &Interpolate::interpolate);
+        clsInterpolate.def("interpolate",
+                           (ndarray::Array<double, 1> (Interpolate::*) (ndarray::Array<double const, 1> const&)
+                               const) &Interpolate::interpolate);
+
+        return mod.ptr();
+    }
+
```

Comment:
I completely agree. I was hoping to finish this ticket this sprint but you and Russel gave a lot of good feedback so I'm going to take another pass through this and make some significant upgrades. If you don't mind I'd like you to have another look in a day or two once it's been updated.